sky/serve/control_plane.py

```diff
@@ -61,41 +74,51 @@ def get_autoscaler_query_interval():
         def get_ready_replicas():
             return {'ready_replicas': self.infra_provider.get_ready_replicas()}
 
-        @self.app.get('/control_plane/get_replica_info')
-        def get_replica_info():
-            return {'replica_info': self.infra_provider.get_replica_info()}
-
-        @self.app.get('/control_plane/get_replica_nums')
-        def get_replica_nums():
-            return {
-                'num_ready_replicas': self.infra_provider.ready_replica_num(),
-                'num_unhealthy_replicas':
-                    self.infra_provider.unhealthy_replica_num(),
-                'num_failed_replicas': self.infra_provider.failed_replica_num()
+        @self.app.get('/control_plane/get_latest_info')
+        def get_latest_info():
+            latest_info = {
+                'replica_info':
+                    self.infra_provider.get_replica_info(verbose=True),
+                'uptime': self.infra_provider.get_uptime(),
             }
+            latest_info = {
+                k: base64.b64encode(pickle.dumps(v)).decode('utf-8')
+                for k, v in latest_info.items()
+            }
+            return latest_info
 
         @self.app.post('/control_plane/terminate')
         def terminate(request: fastapi.Request):
             del request
-            # request_data = request.json()
-            # TODO(tian): Authentication!!!
             logger.info('Terminating service...')
-            self.infra_provider.terminate_replica_fetcher()
             if self.autoscaler is not None:
+                logger.info('Terminate autoscaler monitor...')
                 self.autoscaler.terminate_monitor()
             msg = self.infra_provider.terminate()
+            # Cleanup cloud storage
+            # TODO(tian): move to local serve_down so that we can cleanup
+            # local storage cache as well.
+            task = sky.Task.from_yaml(self.task_yaml)
+            backend = backends.CloudVmRayBackend()
+            backend.teardown_ephemeral_storage(task)
             return {'message': msg}
 
         # Run replica_monitor and autoscaler.monitor (if autoscaler is defined)
         # in separate threads in the background.
         # This should not block the main thread.
-        self.infra_provider.start_replica_fetcher()
+        self.infra_provider.start_replica_prober()
         if self.autoscaler is not None:
             self.autoscaler.start_monitor()
```

Comment:
Actually why is it called monitor? it takes action (autoscaling) right not just "monitoring"

Reply:
now just calling it autoscaler